src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -407,7 +409,7 @@ private void CreateNodeCaches()
                 if (CompilationModuleGroup.ShouldProduceFullVTable(type))
                     return new EagerlyBuiltVTableSliceNode(type);
                 else
-                    return new LazilyBuiltVTableSliceNode(type);
+                    return _vtableSliceProvider.GetSlice(type);
```

Comment:
This feels like a half-step. Why don't we get rid of the use of ShouldProduceFullVTable here? Or is that planned for later work?